src/test/java/selenium/pageobjects/WebLabAssignmentListPage.java

```diff
@@ -0,0 +1,51 @@
+package selenium.pageobjects;
+
+import org.openqa.selenium.By;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class WebLabAssignmentListPage extends BasePageObject {
+
+    public WebLabAssignmentListPage(WebDriver driver, String url) {
+        super(driver, url);
+    }
+
+    @FindBy(className = "majorrow")
+    private List<WebElement> subAssignments;
+
+    public List<String> getSubdirectoryUrls() {
+        return getSubAssignmentUrls("Collection of Assignments");
+    }
+
+    public List<String> getAssignmentUrls() {
+        return getSubAssignmentUrls("Programming question (docker:cse1110-andy)");
+    }
+
+    private List<String> getSubAssignmentUrls(String type) {
+        this.awaitListLoaded();
+
+        return subAssignments.stream()
+                .filter(tr -> tr
+                        .findElements(By.tagName("td"))
+                        .get(0)
+                        .findElement(By.tagName("span"))
+                        .getAttribute("data-original-title")
+                        .equals(type)
+                )
+                .map(tr -> tr
+                        .findElements(By.tagName("td"))
+                        .get(1)
+                        .findElement(By.tagName("a"))
+                        .getAttribute("href")
+                )
+                .collect(Collectors.toList());
+    }
```

Comment:
Is there a better way of doing this via the Page Object pattern?

Reply:
What exactly? This looks good to me.

Reply:
Could we maybe have another class that describes a single `subAssignment` element? Something like the Page Object we have for the entire page, but only for one element with subelements, so that I don'thave to use `.findElements(By.tagName("td")).get(0).findElement(By.tagName("span")).getAttribute("data-original-title")` but instead I could do something like `subAssignment.getAssignmentType()`?

Reply:
Sure, why not?!

Or do you think a simple private method does the job?

Reply:
I added a class named WebLabAssignmentListElement to do this. However, it seems like it's not possible to use `@FindBy` in nested objects without additional libraries, so I created the page object a bit differently than the rest: [WebLabAssignmentListElement.java](https://github.com/cse1110/andy/blob/094e6a0abe9128da6856dbe494d90401b285cdc2/src/test/java/selenium/pageobjects/WebLabAssignmentListElement.java)

Is there a better way to achieve this?